src/chunkserver/heartbeat.cpp

```diff
@@ -491,41 +499,43 @@ int Heartbeat::ExecTask(const HeartbeatResponse& response) {
         case curve::mds::heartbeat::START_SCAN_PEER:
             {
                 ConfigChangeType type;
-                Configuration confTemp;
+                Configuration tmpConf;
                 Peer peer;
                 LogicPoolID poolId = conf.logicalpoolid();
                 CopysetID copysetId = conf.copysetid();
                 int ret = 0;
                 // if copyset happen conf change, can't scan and wait retry
-                if (ret = copyset->GetConfChange(&type,
-                                                &confTemp, &peer) != 0) {
+                if (ret = copyset->GetConfChange(&type, &tmpConf, &peer) != 0) {
```

Comment:
> Because `!=` has higher priority than `=`, so `copyset->GetConfChange(&type, &tmpConf, &peer) != 0` will evulate frist, and its result will pass to `ret`. Is this order what you expect?

fixed